cmd/snap-recovery/partition/mkfs.go

```diff
@@ -24,12 +24,25 @@ import (
 	"github.com/snapcore/snapd/gadget"
 )
 
-func makeFilesystem(node, label, filesystem, content string) error {
+func MakeFilesystems(created map[string]gadget.LaidOutStructure) error {
+	for node, part := range created {
+		if part.VolumeStructure.Filesystem != "" {
+			if err := mkfs(node, part.VolumeStructure.Label, part.VolumeStructure.Filesystem); err != nil {
```

Comment:
I don't like the fact that this creates filesystems in non-deterministic order. Could you please consider sorting them by node name so that this function is entirely deterministic.

Reply:
There will be a followup that will use the gpt partition attributes to mark partitions as ready or in progress. I assume your concern is idempotency? This will be addressed. Or is there another reason why this should be deterministic?

Reply:
In this specific case my concern is determinism. If this is used on millions of devices it should not result in sometimes doing partition A ahead of B and sometimes the other way around.

Reply:
Yeah, I agree with @zyga but a doing that in the next PR sounds good too.